src/main/java/gyro/azure/keyvault/VaultCertificateResource.java

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright 2020, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.azure.keyvault;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import com.microsoft.azure.keyvault.models.CertificateBundle;
+import com.microsoft.azure.keyvault.models.CertificateOperation;
+import com.microsoft.azure.keyvault.requests.CreateCertificateRequest;
+import com.microsoft.azure.management.Azure;
+import com.microsoft.azure.management.keyvault.Vault;
+import gyro.azure.AzureResource;
+import gyro.azure.Copyable;
+import gyro.core.GyroCore;
+import gyro.core.GyroException;
+import gyro.core.GyroUI;
+import gyro.core.Type;
+import gyro.core.Wait;
+import gyro.core.resource.Id;
+import gyro.core.resource.Output;
+import gyro.core.resource.Resource;
+import gyro.core.scope.State;
+import gyro.core.validation.Required;
+
+/**
+ * Creates a vault certificate.
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *     azure::vault-certificate vault-certificate-example
+ *         name: "certificate-example"
+ *         vault: $(azure::vault vault-example)
+ *
+ *         policy
+ *             key-properties
+ *                 exportable: false
+ *                 reuse-key: false
+ *                 size: 2048
+ *                 type: "RSA"
+ *             end
+ *
+ *             lifetime-action
+ *                 action
+ *                     type: "EmailContacts"
+ *                 end
+ *
+ *                 trigger
+ *                     lifetime-percentage: 90
+ *                 end
+ *             end
+ *
+ *             secret-properties
+ *                 content-type: "application/x-pkcs12"
+ *             end
+ *
+ *             x509-properties
+ *                 key-usage: ["digitalSignature", "keyEncipherment"]
+ *                 subject: "CN=a1.com"
+ *                 validity-in-months:  2
+ *                 ekus: ["1.3.6.1.5.5.7.3.1", "1.3.6.1.5.5.7.3.2"]
+ *             end
+ *
+ *             attribute
+ *                 "enabled" : true
+ *                 "expires" : "2020-04-03T15:54:12.000Z"
+ *             end
+ *
+ *             issuer-parameter
+ *                 name: "Self"
+ *             end
+ *         end
+ *     end
+ */
+@Type("vault-certificate")
```

Comment:
This should be `@Type("key-vault-certificate")`. Update any examples/docs as well.

Reply:
Updated and renamed class to append with `key` as well.